views/forgot.go

```diff
@@ -28,73 +28,46 @@ var notification = Notification{
 }
 
 // ForgotFunc handles sending a reset email
-func (v *Views) ForgotFunc(w http.ResponseWriter, r *http.Request) {
+func (v *Views) ForgotFunc(c echo.Context) error {
 	var err error
-	switch r.Method {
+	switch c.Request().Method {
 	case "GET":
-		err = v.template.RenderNoNavsTemplate(w, nil, templates.ForgotTemplate)
-		if err != nil {
-			err = fmt.Errorf("failed to exec tmpl: %w", err)
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-		}
-		return
+		return v.template.RenderTemplate(c.Response().Writer, nil, templates.ForgotTemplate, templates.NoNavType)
 	case "POST":
-		err = r.ParseForm()
+		err = c.Request().ParseForm()
 		if err != nil {
-			http.Error(w, err.Error(), http.StatusBadRequest)
+			http.Error(c.Response(), err.Error(), http.StatusBadRequest)
 		}
-		u := user.User{Email: r.Form.Get("email")}
+		u := user.User{Email: c.Request().Form.Get("email")}
 
 		if u.Email == "" {
-			err = v.template.RenderNoNavsTemplate(w, nil, templates.ForgotTemplate)
-			if err != nil {
-				err = fmt.Errorf("failed to exec tmpl: %w", err)
-				http.Error(w, err.Error(), http.StatusInternalServerError)
-			}
-			return
+			return v.template.RenderTemplate(c.Response(), nil, templates.ForgotTemplate, templates.NoNavType)
 		}
-		// Get user1 and check if it exists
-		user1, err := v.user.GetUser(r.Context(), u)
+		// Get user and check if it exists
+		user1, err := v.user.GetUser(c.Request().Context(), u)
 		if err != nil {
 			// User doesn't exist, we'll pretend they've got an email
 			log.Printf("request for reset on unknown email \"%s\"", user1.Email)
-			err = v.template.RenderNoNavsTemplate(w, notification, templates.NotificationTemplate)
-			if err != nil {
-				err = fmt.Errorf("failed to exec template: %w", err)
-				http.Error(w, err.Error(), http.StatusInternalServerError)
-			}
-			return
+			return v.template.RenderTemplate(c.Response(), notification, templates.NotificationTemplate, templates.NoNavType)
 		}
 		url := uuid.NewString()
 		v.cache.Set(url, user1.UserID, cache.DefaultExpiration)
 
 		// Valid request, send email with reset code
-		if v.mailer.Enabled {
-			v.mailer, err = mail.NewMailer(mail.Config{
+		if v.mailer != nil {
+			v.mailer = mail.NewMailer(mail.Config{
 				Host:       v.conf.Mail.Host,
 				Port:       v.conf.Mail.Port,
 				Username:   v.conf.Mail.Username,
 				Password:   v.conf.Mail.Password,
 				DomainName: v.conf.DomainName,
 			})
```

Comment:
Why are we creating a new mailer here when it should have already been created in `main.go`? How likely is it that `v.Conf.Mail` changes in between startup and reaching this point?

Reply:
Funny you mention that, if it gets disconnected or times out, yes I know there is a keep alive but doesn't appear to keep alive for too long and therefore needs to be recreated, I ran into that issue with STV and also here, regardless of what is changed after startup, the values at startup will be used because it only gets the env values once

Reply:
Hang on, is it keeping a SMTP connection open the whole time? That seems wasteful to me when we could go days without sending a single email.

If that's the case I'd consider reworking `Mailer` to connect, send an email, and disconnect.

Reply:
I'll rework it

Reply:
This has been redone